flytekit/annotated/launch_plan.py

```diff
@@ -0,0 +1,233 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional
+
+from flytekit import logger
+from flytekit.annotated import workflow as _annotated_workflow
+from flytekit.annotated.context_manager import FlyteContext, FlyteEntities
+from flytekit.annotated.interface import Interface, transform_inputs_to_parameters
+from flytekit.annotated.type_engine import TypeEngine
+from flytekit.common.launch_plan import SdkLaunchPlan
+from flytekit.models import common as _common_models
+from flytekit.models import interface as _interface_models
+from flytekit.models import launch_plan as _launch_plan_models
+from flytekit.models import literals as _literal_models
+from flytekit.models import schedule as _schedule_model
+from flytekit.models.core import identifier as _identifier_model
+
+
+class LaunchPlan(object):
+    # The reason we cache is simply because users may get the default launch plan twice for a single Workflow. We
+    # don't want to create two defaults, could be confusing.
+    CACHE = {}
+
+    @staticmethod
+    def native_kwargs_to_literal_map(
```

Comment:
can we add this method to TypeEngine ?

Reply:
Interface already depends on TypeEngine.  Don't want a circular dependency.  How about we add to Interface?

Reply:
in that case maybe just a helper method?